### ISSUE TITLE
Support confidential nodepools

### ISSUE THREAD

**Issue by jawnsy** [terraform-google-modules/terraform-google-kubernetes-engine#1386]
### TL;DR

We currently support confidential clusters (a cluster-level setting that enforces that all nodepools are confidential VMs), but do not support the setting at a per-nodepool level. We could add a new node_pool setting (e.g. `enable_confidential_nodes`) to enable this on a per-nodepool basis, similar to the `enable_secure_boot` setting)

### Additional information

[GKE supports mixed clusters](https://cloud.google.com/blog/products/identity-security/announcing-general-availability-of-confidential-gke-nodes) where some nodepools are confidential instances (n2d or c2d) but others run on other instances (n1, n2, c2, t2d), but the submodule does not support this.

**Comment by bharathkkb** [comment]
@jawnsy happy to review a PR exposing this!

**Comment by maksym-kursin** [comment]
It seems impossible to implement it right now because a confidential node is not supported by [container_node_pool](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/container_node_pool) resource.

Here is the related provider issue: https://github.com/hashicorp/terraform-provider-google/issues/13127

**Comment by DrFaust92** [comment]
@apeabody, I think this can be closed as its supported already

**Comment by apeabody** [comment]
Thanks, added in https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/pull/1815

**Comment by redoak666** [comment]
#1815 does not include support for nodepool level enablement of confidential nodes, only cluster level is supported
#1756 is the correct fix, but NOT merged

**Comment by redoak666** [comment]
as of today, the latest release v32.0.4 lacks of capability of nodepool level enablement of confidential nodes

**Comment by redoak666** [comment]
Since v30 is the release that confidential node goes GA, can this confidential nodepool support backported to v30?
